FirebaseVertexAI/Sources/Logging.swift

```diff
@@ -18,13 +18,19 @@ import OSLog
 @available(iOS 15.0, macOS 11.0, macCatalyst 15.0, *)
 struct Logging {
   /// Subsystem that should be used for all Loggers.
-  static let subsystem = "com.google.generative-ai"
+  static let subsystem = "com.google.firebase.vertex-ai"
```

Comment:
It seems that subsystem is redundant with `[FirebaseVertexAI]`?   I'm fine to leave that as a future question.

Reply:
Thanks for bringing it up, @paulb777. It is largely redundant given that a Firebase dev would need to filter by tag (like `[FirebaseVertexAI]`) since the rest of our SDKs print like this. However, if we were to migrate everything to OSLog in the future it would actually be `[FirebaseVertexAI]` that would be redundant (along with `-FIRDebugEnabled`). Rather than setting a log level as a launch argument in Xcode, with OSLog it's easy to filter by levels and subsystems in the Xcode UI, e.g.:
![Xcode OSLog Screenshot](https://github.com/firebase/firebase-ios-sdk/assets/3010484/959221ba-e693-4505-8d58-64ada88b24dc)
In the above, I've enabled the option to display the subsystem and to filter by it, as well as log level. Definitely more flexible to tweak these as needed without having to restart the app. These options are also available in the Console app.

For now I'll leave it in (alternative is an empty string, like we did for `category`, since the parameter is non-optional) but we can adjust later or migrate to `FirebaseLogger`.